main/src/mill/main/MainModule.scala

```diff
@@ -177,4 +182,42 @@ trait MainModule extends mill.Module{
       }
     }
   }
+
+  /**
+    * Deletes the given targets from the out directory. Providing no targets
+    * will clean everything.
+    */
+  def clean(evaluator: Evaluator[Any], targets: String*) = mill.T.command {
+    val rootDir = ammonite.ops.pwd / OutDir
+
+    val KeepPattern = "(mill-worker-[0-9]+)".r.anchored
+    
+    def keepPath(path: Path) = path.segments.lastOption match {
+      case Some(KeepPattern(_)) => true
+      case _ => false
+    }
+
+    val pathsToRemove =
+      if (targets.isEmpty)
+        Right(ammonite.ops.ls(rootDir).filterNot(keepPath))
```

Comment:
Turns out the error on "filter out" was caused by deleting `out/mill-worker-1`. For now I have hard-coded a filter for anything like `out/mill-worker-N`, but:
- I'm not sure this is what we'd want?
- If we do, I don't know if there is a better way to filter those folders ?